sdk/scheduler/src/test/java/org/apache/mesos/offer/constrain/MarathonConstraintParserTest.java

```diff
@@ -0,0 +1,224 @@
+
+    @Test
+    public void testSplitConstraints() throws IOException {
+        assertEquals(Arrays.asList(Arrays.asList("")),
+                MarathonConstraintParser.splitConstraints(""));
+        assertEquals(Arrays.asList(Arrays.asList("a")),
+                MarathonConstraintParser.splitConstraints("a"));
+        assertEquals(Arrays.asList(Arrays.asList("a", "b", "c")),
+                MarathonConstraintParser.splitConstraints(unescape("['a', 'b', 'c']")));
+        assertEquals(Arrays.asList(Arrays.asList("a", "b", "c")),
+                MarathonConstraintParser.splitConstraints(unescape("[['a', 'b', 'c']]")));
+        assertEquals(Arrays.asList(Arrays.asList("a", "b", "c"), Arrays.asList("d", "e")),
+                MarathonConstraintParser.splitConstraints(unescape("[['a', 'b', 'c'], ['d', 'e']]")));
+        assertEquals(Arrays.asList(Arrays.asList("a"), Collections.emptyList()),
+                MarathonConstraintParser.splitConstraints(unescape("[['a'], []]")));
+        assertEquals(Arrays.asList(Arrays.asList("a", "b", "c"), Arrays.asList("d", "e"), Arrays.asList("f", "g", "")),
+                MarathonConstraintParser.splitConstraints("a:b:c,d:e,f:g:"));
+        assertEquals(Arrays.asList(Arrays.asList("a", "b", "c"), Arrays.asList("", "d", "e"), Arrays.asList("", "f", "")),
+                MarathonConstraintParser.splitConstraints("a:b:c,:d:e,:f:"));
+        assertEquals(Arrays.asList(Arrays.asList("a", "b", "c"), Arrays.asList("", "d", "e"), Arrays.asList("", "f", "")),
+                MarathonConstraintParser.splitConstraints(" a : b : c , : d : e , : f : "));
+        assertEquals(Arrays.asList(Arrays.asList("", "", ""), Arrays.asList("", "")),
+                MarathonConstraintParser.splitConstraints("::,:"));
+    }
+
+    @Test
+    public void testUniqueOperator() throws IOException {
+        // marathon doc:
+        String constraintStr = MarathonConstraintParser.parse(unescape("[['hostname', 'UNIQUE']]")).toString();
+        assertEquals("MaxPerHostnameRule{max=1, task_filter=AnyMatcher{}}", constraintStr);
+        assertEquals(constraintStr, MarathonConstraintParser.parse(unescape("['hostname', 'UNIQUE']")).toString());
+        assertEquals(constraintStr, MarathonConstraintParser.parse("hostname:UNIQUE").toString());
+
+        constraintStr = MarathonConstraintParser.parse(unescape("[['rack_id', 'UNIQUE']]")).toString();
+        assertEquals("AndRule{rules=[AttributeRule{matcher=RegexMatcher{pattern='rack_id:.*'}}, " +
+                "MaxPerAttributeRule{max=1, matcher=RegexMatcher{pattern='rack_id:.*'}, task_filter=AnyMatcher{}}]}", constraintStr);
+        assertEquals(constraintStr, MarathonConstraintParser.parse(unescape("['rack_id', 'UNIQUE']")).toString());
+        assertEquals(constraintStr, MarathonConstraintParser.parse("rack_id:UNIQUE").toString());
+    }
+
+    @Test
+    public void testClusterOperator() throws IOException {
+        // marathon doc:
```

Comment:
These were meant to highlight the cases where the tests are against something from the marathon doc at https://mesosphere.github.io/marathon/docs/constraints.html. Updated them to be clearer about this.